packages/neon/neon/lib/src/blocs/unified_search.dart

```diff
@@ -0,0 +1,161 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+import 'package:neon/models.dart';
+import 'package:neon/src/bloc/bloc.dart';
+import 'package:neon/src/bloc/result.dart';
+import 'package:neon/src/blocs/apps.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:rxdart/rxdart.dart';
+
+abstract interface class UnifiedSearchBlocEvents {
+  void search(final String term);
+
+  void enable();
+
+  void disable();
+}
+
+abstract interface class UnifiedSearchBlocStates {
+  BehaviorSubject<bool> get enabled;
+
+  BehaviorSubject<Result<Map<CoreUnifiedSearchProvider, Result<CoreUnifiedSearchResult>>?>> get results;
+}
+
+@internal
+class UnifiedSearchBloc extends InteractiveBloc implements UnifiedSearchBlocEvents, UnifiedSearchBlocStates {
+  UnifiedSearchBloc(
+    this._appsBloc,
+    this._account,
+  ) {
+    _appsBloc.activeApp.listen((final _) {
+      if (enabled.value) {
+        disable();
+      }
+    });
+  }
+
+  final AppsBloc _appsBloc;
+  final Account _account;
+  String _term = '';
+
+  @override
+  BehaviorSubject<bool> enabled = BehaviorSubject.seeded(false);
+
+  @override
+  BehaviorSubject<Result<Map<CoreUnifiedSearchProvider, Result<CoreUnifiedSearchResult>>?>> results =
+      BehaviorSubject.seeded(Result.success(null));
+
+  @override
+  void dispose() {
+    unawaited(enabled.close());
+    unawaited(results.close());
+    super.dispose();
+  }
+
+  @override
+  Future refresh() async {
+    await _search();
+  }
+
+  @override
+  Future search(final String term) async {
+    _term = term.trim();
+    await _search();
+  }
+
+  @override
+  void enable() {
+    enabled.add(true);
+  }
+
+  @override
+  void disable() {
+    enabled.add(false);
+    results.add(Result.success(null));
+    _term = '';
+  }
+
+  Future _search() async {
+    if (_term.isEmpty) {
+      results.add(Result.success(null));
+      return;
+    }
+
+    try {
+      results.add(results.value.asLoading());
+      final providers = (await _account.client.core.unifiedSearch.getProviders()).ocs.data;
+      results.add(
+        Result.success({
+          for (final provider in providers) ...{
+            provider: Result.loading(),
+          },
+        }),
```

Comment:
Please use a generator function here

Reply:
Make a `sync*` annotated function that returns `Iterable<MapEntry> and `yield` the values you want to add.

Then you can call the mentioned Map.fromEntries.

This should require creating less maps and "changing" the pointer of the enties.

Maybe https://www.geeksforgeeks.org/generators-in-dart/ is helpful.

Reply:
Ah, I only knew async* generators, this makes sense :)